### PR TITLE
ArraySetLength lib function

### LINKED ISSUE #514: array.length = 0 be translated to #array = 0
array.length = 0 be translated to #array = 0
typescript:
` 
let list = [1,2,3,4,5];
list.length = 0; //clear the array
`
lua:
`
local list = {1, 2, 3, 4, 5};
#list = 0;
`

### PR DESCRIPTION
**Pull request by tomblind**
fixes #514

### CODE REVIEW

File: src/lualib/ArraySetLength.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function __TS__ArraySetLength<T>(this: void, arr: T[], length: number): number {
+    if (length < 0) {
```
Comment (ark120202) [inline src/lualib/ArraySetLength.ts]: In JS it fails for any non-integers (32 bit), which also includes `.length = 0.1`, `.length = Infinity` and `.length = NaN`.

**Comment by tomblind** [comment]
Updated to handle NaN, Infinity, and non-integers

File: src/lualib/ArraySetLength.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+function __TS__ArraySetLength<T>(this: void, arr: T[], length: number): number {
+    if (length < 0
+        || length !== length // NaN
+        || length === (1 / 0) // Infinity
+        || length === (-1 / 0) // -Infinity
```
Comment (ark120202) [inline src/lualib/ArraySetLength.ts]: I think `length < 0` should be enough to catch it

Comment (tomblind) [reply src/lualib/ArraySetLength.ts]: good point

**Pull request merged**